lib/logstorage/storage.go

```diff
@@ -141,6 +141,22 @@ type Storage struct {
 	//
 	// It reduces the load on persistent storage during querying by _stream:{...} filter.
 	filterStreamCache *cache
+
+	// minKeptDay is set ONLY after the disk-space watcher removes recent partitions.
+	// It marks the earliest day still kept; older days are rejected during ingestion
+	// to avoid re-creating partitions just purged for space.
+	minKeptDay atomic.Int64
+}
+
+// setMinKeptDayFromPartitions updates minKeptDay based on the current list of partitions.
+// It must be called while s.partitions reflects the up-to-date state. The caller may hold
+// or not hold partitionsLock depending on the context — the function itself doesn't lock.
+func (s *Storage) setMinKeptDayFromPartitions() {
```

Comment:
It is better from readability and maintainability PoV to inline manually this function at the single call place. This removes an additional level of indirection and simplifies reading the code without the need to jump to this function body.